cirq/ops/pauli_string.py

```diff
@@ -22,43 +23,85 @@
 import numpy as np
 
 from cirq import value, protocols, linalg
-from cirq.ops import (
-    global_phase_op,
-    raw_types,
-    gate_operation,
-    common_gates,
-    pauli_gates,
-    clifford_gate,
-    pauli_interaction_gate,
-)
+from cirq._compat import deprecated
+from cirq.ops import (global_phase_op, raw_types, gate_operation, common_gates,
+                      pauli_gates, clifford_gate, pauli_interaction_gate)
+
+if TYPE_CHECKING:
+    import cirq
+
+# A value that can be unambiguously converted into a `cirq.PauliString`.
+PAULI_STRING_LIKE = Union[
+    complex, 'cirq.OP_TREE',
+    Mapping['cirq.Qid', Union['cirq.Pauli', 'cirq.IdentityGate']],
+    Iterable,  # of PAULI_STRING_LIKE, but mypy doesn't do recursive types yet.
```

Comment:
Can do recursion with protocols if you want to preserve more type info, as in #2140.